src/OpenTelemetry.AutoInstrumentation.Native/logger_impl.h

```diff
@@ -90,42 +86,70 @@ std::string LoggerImpl<TLoggerPolicy>::GetLogPath(const std::string& file_name_s
 template <typename TLoggerPolicy>
 LoggerImpl<TLoggerPolicy>::LoggerImpl()
 {
-    m_debug_logging_enabled = false;
-
     spdlog::set_error_handler([](const std::string& msg) {
         // By writing into the stderr was changing the behavior in a CI scenario.
         // There's not a good way to report errors when trying to create the log file.
         // But we never should be changing the normal behavior of an app.
         // std::cerr << "LoggerImpl Handler: " << msg << std::endl;
     });
 
-    spdlog::flush_every(std::chrono::seconds(3));
-
-    static auto current_process_name = ToString(GetCurrentProcessName());
-    static auto current_process_id = GetPID();
-    static auto current_process_without_extension =
-        current_process_name.substr(0, current_process_name.find_last_of("."));
-
-    static auto file_name_suffix = "-" + current_process_without_extension + "-" + std::to_string(current_process_id);
+    static auto configured_log_level = GetEnvironmentValue(environment::log_level);
 
-    try
+    if (configured_log_level == WStr("none"))
     {
-        m_fileout = spdlog::rotating_logger_mt("Logger", GetLogPath(file_name_suffix), 1048576 * 5, 10);
-    }
-    catch (...)
-    {
-        // By writing into the stderr was changing the behavior in a CI scenario.
-        // There's not a good way to report errors when trying to create the log file.
-        // But we never should be changing the normal behavior of an app.
-        // std::cerr << "LoggerImpl Handler: Error creating native log file." << std::endl;
         m_fileout = spdlog::null_logger_mt("LoggerImpl");
     }
+    else
+    {
+        auto log_level = spdlog::level::info;
 
-    m_fileout->set_level(spdlog::level::debug);
+        if (configured_log_level == WStr("error"))
```

Comment:
How about extracting the `none` `debug` `info` `warn` `error` constants?

Reply:
done in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2288/commits/3472adb5d87bbab5d8a433bf83c910fd3b2d8bda